agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFileSender.java

```diff
@@ -56,12 +57,12 @@ private LocalFileSender(LocalFileLoader localFileLoader, TelemetryChannel teleme
 
   @Override
   public void run() {
-    // TODO (heya) should this loop more aggressively until it reads all telemetry from disk?
-    //  or is it intentional to only send one batch per 30 seconds?
     try {
       ByteBuffer buffer = localFileLoader.loadTelemetriesFromDisk();
       if (buffer != null) {
-        telemetryChannel.sendRawBytes(buffer);
+        CompletableResultCode resultCode = telemetryChannel.sendRawBytes(buffer);
+        resultCode.join(5, TimeUnit.SECONDS); // wait for request to be completed.
```

Comment:
avg statsbeat request is less than 1 second.  is 5 too long and that is also depended on the network speed.